### PR TITLE
feat: enhanced visualize.py with NiceGUI

### LINKED ISSUE #73: Refactor visualize.py with NiceGUI
https://github.com/OpenAdaptAI/OpenAdapt/blob/main/openadapt/visualize.py should be rewritten, maybe with https://nicegui.io/ , maybe something else.

We would also like to display performance information (see https://github.com/MLDSAI/OpenAdapt/issues/85)

### PR DESCRIPTION
**Pull request by 0dm**
new visualization:

image
image
(buttons to view bigger / save plot)

now says if scrubbing:
image

events:
image
searching:
image
image

environment variables:
```
    "VISUALIZE_DARK_MODE": False,
    "RUN_NATIVELY": False,
    "DENSE_TREES": True,
    "NO_ANIMATIONS": False,
    "EXPAND_ALL": False,
    "MAX_TABLE_CHILDREN": 10,
```
outside, of visualize, I have edited utils.plot_performance with some new default parameters:
```py
def plot_performance(
    recording_timestamp: float = None,
    view_file: bool = True,
    save_file: bool = True,
    dark_mode: bool = False,
) -> str:
```
I have moved the old visualize.py to openadapt/deprecated. `python -m openadapt.deprecated.visualize` just 
as a fallback (nicegui breaks / compatibility)

this is a pretty big PR there's a lot of code (some of it might look a little strange..)

### CODE REVIEW

**Comment by 0dm** [comment]
image

File: openadapt/visualize1.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    configure_logging,
+    display_event,
+    image2utf8,
+    row2dict,
+    rows2dicts,
+    plot_performance,
+)
+
+SCRUB = config.SCRUB_ENABLED
+if SCRUB:
```
Comment (abrichr) [inline openadapt/visualize1.py:21]: how about just `if config.SCRUB_ENABLED`?

File: openadapt/visualize1.py [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    # too many warnings from scrubbing
+    __import__("warnings").filterwarnings("ignore", category=DeprecationWarning)
```
Comment (abrichr) [inline openadapt/visualize1.py:23]: I think we can remove this once https://github.com/OpenAdaptAI/OpenAdapt/pull/389 is merged right?

File: openadapt/visualize1.py [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+
+
+def create_tree(tree_dict, max_children=MAX_TABLE_CHILDREN):
+    tree_data = []
+    for key, value in tree_dict.items():
+        if value in EMPTY:
+            continue
+        node = {
+            "id": str(key)
+            + f"{(': '  + str(value)) if not isinstance(value, (dict, list)) else ''}"
```
Comment (abrichr) [inline openadapt/visualize1.py:40]: Seems like the f-string is unused here. Can you please re-write this to use the f-string?

File: openadapt/visualize1.py (hunk 2 above)
Comment (0dm) [reply openadapt/visualize1.py:23]: possibly, I think there'd need to be another change to send those warnings to loguru first though.

File: openadapt/visualize1.py [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+            else:
+                node["children"] = create_tree({i: v for i, v in enumerate(value)})
+        tree_data.append(node)
+    return tree_data
+
+
+def set_tree_props(tree):
+    tree._props["dense"] = True
+    tree._props["no-transition"] = True
+    # tree._props["default-expand-all"] = True
```
Comment (abrichr) [inline openadapt/visualize1.py:59]: How about making this configurable?

File: openadapt/visualize1.py (hunk 1 above)
Comment (0dm) [reply openadapt/visualize1.py:21]: I declare SCRUB because it gets checked quite often in the main function.

File: openadapt/visualize1.py (hunk 4 above)
Comment (0dm) [reply openadapt/visualize1.py:59]: Yeah I'll use this to function to fetch the newvariables from openadapt.config 👍 just had it like this for testing

**Comment by 0dm** [comment]
@abrichr this is ready for review

**Comment by 0dm** [comment]
note:
diff & mask are not being used --I will find a way to integrate them soon

**Pull request merged**